src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,381 @@
+  /**
+   * The upper scrollable container
+   */
+  upperScrollableContainer: HTMLElement;
+  /**
+   * The original css of the sticky element, used to reset the sticky element
+   * when it is being unstuck
+   */
+  originalCss: any;
+  /**
+   * 'getBoundingClientRect().top' of CdkStickyRegion of current sticky header.
+   * It is used with '_scrollFinish' to judge whether the current header
+   * need to be stuck.
+   */
+  private _containerStart: number;
+  /**
+   * It is 'The bottom of CdkStickyRegion of current sticky header - the height
+   * of current header', which is used with '_containerStart' to judge whether the current header
+   * need to be stuck.
+   */
+  private _scrollFinish: number;
+  /**
+   * The width of the sticky-header when it is stuck.
+   */
+  private _scrollingWidth: number;
+
+  constructor(_element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentReg: CdkStickyRegion) {
+    this.element = _element.nativeElement;
+    this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+
+    this.detectBrowser();
+  }
+
+  ngAfterViewInit(): void {
+    this.stickyParent = this.parentReg != null ?
+      this.parentReg._elementRef.nativeElement : this.element.parentElement;
+    this.originalCss = this.generateCssStyle(this.getCssValue(this.element, 'zIndex'),
+      this.getCssValue(this.element, 'position'), this.getCssValue(this.element, 'top'),
+      this.getCssValue(this.element, 'right'), this.getCssValue(this.element, 'left'),
+      this.getCssValue(this.element, 'bottom'), this.getCssValue(this.element, 'width'));
+    if(this.isIE == true) {
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.upperScrollableContainer.removeEventListener('scroll', this._onScrollBind);
+    this.upperScrollableContainer.removeEventListener('resize', this._onResizeBind);
+    this.upperScrollableContainer.removeEventListener('touchmove', this._onTouchMoveBind);
+  }
+
+  /**
+   * Use 'navigator.appVersion' to detect current browser. According to the "Position:sticky
+   * Browser compatibility" in "https://developer.mozilla.org/en-US/docs/Web/CSS/position".
+   *
+   * For Desktop: Sticky positioning works well on Chrome, Edge, Firefox and Opera. And can
+   * also work well on Safari with a "-webkit-" prefix. It only does not work on IE.
+   *
+   * For Mobile: Sticky positioning works well on Android Webview, Chrome for Android, Edge,
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   * So the detectBrowser() function detects the current browser version. If it is about iPhone
+   * safari, set the style.position as '-webkit-sticky'. If it is an IE or an Tablet, use the
+   * original implementation. And for other circumstances, use 'position: 'sticky''.
+   */
+  detectBrowser(): void {
```

Comment:
Use what modernizr does on compatibility instead of get the browser version directly.
Added new functions:  `getSupportList()` and `setStrategyAccordingToCompatibility()`